test/end2end_test.go

```diff
@@ -85,29 +85,29 @@ func (s *testServer) EmptyCall(ctx context.Context, in *testpb.Empty) (*testpb.E
 	return new(testpb.Empty), nil
 }
 
-func newPayload(t testpb.PayloadType, size int32) *testpb.Payload {
+func newPayload(t testpb.PayloadType, size int32) (*testpb.Payload, error) {
 	if size < 0 {
-		grpclog.Fatalf("Requested a response with invalid length %d", size)
+		return nil, fmt.Errorf("Requested a response with invalid length %d", size)
 	}
 	body := make([]byte, size)
 	switch t {
 	case testpb.PayloadType_COMPRESSABLE:
 	case testpb.PayloadType_UNCOMPRESSABLE:
-		grpclog.Fatalf("PayloadType UNCOMPRESSABLE is not supported")
+		return nil, errors.New("PayloadType UNCOMPRESSABLE is not supported")
```

Comment:
errors.New doesn't interpolate arguments, while fmt.Errorf does, so they are semantically different. I could use fmt.Errof everywhere if you feel strongly, but it's somewhat less correct.